test/cri-containerd/removepodsandbox_test.go

```diff
@@ -0,0 +1,171 @@
+// +build functional
+
+package cri_containerd
+
+import (
+	"context"
+	"testing"
+
+	runtime "k8s.io/cri-api/pkg/apis/runtime/v1alpha2"
+)
+
+type TestConfig struct {
+	name             string
+	containerName    string
+	requiredFeatures []string
+	runtimeHandler   string
+	sandboxImage     string
+	containerImage   string
+	cmd              []string
+}
+
+func getRunPodSandboxRequest(t *testing.T, runtimeHandler string) runtime.RunPodSandboxRequest {
+	return runtime.RunPodSandboxRequest{
+		Config: &runtime.PodSandboxConfig{
+			Metadata: &runtime.PodSandboxMetadata{
+				Name:      t.Name(),
+				Uid:       "0",
+				Namespace: testNamespace,
+			},
+		},
+		RuntimeHandler: runtimeHandler,
+	}
+}
+
+// Utility function to run tests for a sandbox pod without any containers
+func runPodSandboxTestWithoutPodStop(t *testing.T, request *runtime.RunPodSandboxRequest) {
+	client := newTestRuntimeClient(t)
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	podId := runPodSandbox(t, client, ctx, request)
+	removePodSandbox(t, client, ctx, podId)
+
+	r, _ := client.PodSandboxStatus(ctx, &runtime.PodSandboxStatusRequest{
+		PodSandboxId: podId,
+	})
+
+	if r != nil {
+		t.Fatalf("Pod should've been removed %s", r)
+	}
```

Comment:
I would prefer we check instead that there is no error here and include the error in the fatal message if there is one.

Reply:
so, the status actually returns "not found" error and I want to make sure that this is the case. if you can suggest a better way of asserting that, it'd be nice!

Reply:
Ah, I got that backward in my previous comment, sorry about that! You could check both the error and status match what you expect if that works